### PR TITLE
doc: add signature documents

### PR DESCRIPTION
**Pull request by binbin-li**
### What?

Add docs to signature module.

Signed-off-by: Binbin Li <libinbin@microsoft.com>

### CODE REVIEW

File: signature/envelope.go [hunk 1]
```diff
@@ -20,22 +20,25 @@ type Envelope interface {
...
+// NewEnvelopeFunc defines a function to create a new Envelope.
 type NewEnvelopeFunc func() Envelope
 
-// ParseEnvelopeFunc defines a function to create a new Envelope with given
-// envelope bytes
+// ParseEnvelopeFunc defines a function to create a new Envelope by given
+// envelope bytes.
 type ParseEnvelopeFunc func([]byte) (Envelope, error)
 
+// envelopeFunc wraps functions to create new envelopes
```
Comment (chloeyin) [inline signature/envelope.go:30]:
```suggestion
// envelopeFunc wraps functions to create and parse new envelopes
```

File: signature/envelope.go [hunk 2]
```diff
@@ -20,22 +20,25 @@ type Envelope interface {
...
 type envelopeFunc struct {
 	newFunc   NewEnvelopeFunc
 	parseFunc ParseEnvelopeFunc
 }
 
+// envelopeFuncs maps envelope media type to corresponding constructors
```
Comment (chloeyin) [inline signature/envelope.go:36]:
```suggestion
// envelopeFuncs maps envelope media type to corresponding constructors and parsers.
```

File: signature/envelope.go (hunk 1 above)
Comment (binbin-li) [reply signature/envelope.go:30]: thanks for catching!

File: signature/envelope.go (hunk 2 above)
Comment (binbin-li) [reply signature/envelope.go:36]: thanks for catching!

File: signature/internal/base/envelope.go [hunk 3]
```diff
@@ -45,11 +51,13 @@ func (e *Envelope) Verify() (*signature.Payload, *signature.SignerInfo, error) {
 		return nil, nil, err
 	}
 
+	// core verify process
```
Comment (chloeyin) [inline signature/internal/base/envelope.go:54]:
```suggestion
	// core verify process.
```
Nit

File: signature/internal/base/envelope.go [hunk 4]
```diff
@@ -45,11 +51,13 @@ func (e *Envelope) Verify() (*signature.Payload, *signature.SignerInfo, error) {
...
 	payload, signerInfo, err := e.Envelope.Verify()
 	if err != nil {
 		return nil, nil, err
 	}
 
+	// validation after the core verify process
```
Comment (chloeyin) [inline signature/internal/base/envelope.go:60]:
```suggestion
	// validation after the core verify process.
```

File: signature/envelope.go [hunk 5]
```diff
@@ -20,22 +20,25 @@ type Envelope interface {
 	SignerInfo() (*SignerInfo, error)
 }
 
-// NewEnvelopeFunc defines a function to create a new Envelope
+// NewEnvelopeFunc defines a function to create a new Envelope.
 type NewEnvelopeFunc func() Envelope
 
-// ParseEnvelopeFunc defines a function to create a new Envelope with given
-// envelope bytes
+// ParseEnvelopeFunc defines a function to create a new Envelope by given
```
Comment (chloeyin) [inline signature/envelope.go:26]: Not a native speaker, just wondering which is better, `create a new Envelope by give envelope bytes` or ``create a new Envelope from give envelope bytes`` :rofl:

File: signature/types.go [hunk 6]
```diff
@@ -5,64 +5,117 @@ import (
...
+	// additional signed attributes in the signature envelope.
 	ExtendedAttributes []Attribute
 }
 
-// UnsignedAttributes represents unsigned metadata in the Signature envelope
+// UnsignedAttributes represents unsigned metadata in the Signature envelope.
+// Reference: https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#unsigned-attributes
 type UnsignedAttributes struct {
+	// SigningAgent provides the identifier of the software (e.g. Notation) that
+	// produced the signature on behalf of the user.
```
Comment (chloeyin) [inline signature/types.go:42]:
```suggestion
        // SigningAgent provides the identifier of the software (e.g., Notation) which
	// produces the signature on behalf of the user.
```

File: signature/envelope.go (hunk 5 above)
Comment (binbin-li) [reply signature/envelope.go:26]: good question. I looked through some repos, `ParseEnvelopeFunc defines a function that takes bytes to create an Envelope` might make more sense.

**Review by chloeyin — APPROVED** [review]
LGTM

**Pull request merged**